projects/Mallard/src/hooks/use-overlay-animation.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useRef, useState } from 'react';
+import { Animated } from 'react-native';
+
+const useOverlayAnimation = (shouldAnimate: boolean) => {
+	const fadeAnim = useRef(new Animated.Value(0)).current;
+
+	const [showOverlay, setShowOverlay] = useState(false);
+
+	React.useEffect(() => {
+		if (shouldAnimate) {
+			setShowOverlay(true);
+			Animated.timing(fadeAnim, {
+				toValue: 0.5,
+				duration: 400,
+				delay: 0,
+			}).start();
+		} else {
+			Animated.timing(fadeAnim, {
+				toValue: 0,
+				duration: 400,
+				delay: 250,
```

Comment:
Not sure if we need a delay on the fade out, feels a little slow.

Reply:
Cool sped it up a little.